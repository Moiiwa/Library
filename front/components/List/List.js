import React from 'react'
import { Link } from "react-router-dom";

class List extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {

        return (
            <ul id="list">
                {this.props.list ?
                    this.props.list.map((el) => (
                        <li key={el.id}>
                            <Link to={`/book/${el.id}`}>{el.title}</Link>
                        </li>
                    ))
                    :
                    null
                }
            </ul>
        );
    }
}

export default List;