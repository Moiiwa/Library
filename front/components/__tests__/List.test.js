import React from 'react';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import List from '../List/List';

Enzyme.configure({ adapter: new Adapter() });

describe('List', () => {

    let mockSellingStatus;
    let mockRentingStatus;

    beforeEach(() => {
        mockSellingStatus = jest.fn();
        mockRentingStatus = jest.fn()
    });

    it('renders correctly with props', () => {
        const props = {
            list: [{ id: 1, title: "title", author: "author", owner: "owner" }]
        }
        const wrapper = shallow(<List {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('works without props', () => {
        const props = {
            list: null
        }
        const wrapper = shallow(<List {...props} />).find('.table');
        expect(wrapper).toEqual({});
    });

    it('update selling status', () => {
        const wrapper = shallow(<List />);
        wrapper.setState({
            books: [{
                id: 2,
                title: "Cell",
                author: "/authors/OL2162284A",
                owner: "anna",
                sellingStatus: false
            }]
        });
        const instance = wrapper.instance()
        const changeSellingStatusMock = jest.spyOn(instance, 'changeSellingStatus')
        instance.forceUpdate();

        const checkbox = wrapper.find('#sell')
        console.debug(checkbox)
        checkbox.simulate('change', { target: { checked: true } });
        expect(changeSellingStatusMock).toHaveBeenCalled()
    });

    it('update sharing status', () => {
        const wrapper = shallow(<List />);
        wrapper.setState({
            books: [{
                id: 2,
                title: "Cell",
                author: "/authors/OL2162284A",
                owner: "anna",
                rentingStatus: false
            }]
        });
        const instance = wrapper.instance()
        const changeSharingStatusMock = jest.spyOn(instance, 'changeSharingStatus')
        instance.forceUpdate();

        const checkbox = wrapper.find('#share')
        console.debug(checkbox)
        checkbox.simulate('change', { target: { checked: true } });
        expect(changeSharingStatusMock).toHaveBeenCalled()
    });
});