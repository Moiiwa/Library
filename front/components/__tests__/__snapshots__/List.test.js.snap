// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly with props 1`] = `
<ul
  id="list"
>
  <li
    key="1"
  >
    <Link
      replace={false}
      to="/book/1"
    >
      title
    </Link>
  </li>
</ul>
`;
